lib/common/time.go

```diff
@@ -17,3 +29,106 @@ func NowISO8601() string {
 func ParseISO8601(s string) (time.Time, error) {
 	return time.Parse(TIMEFORMAT_ISO8601, s)
 }
+
+type TimeSync struct {
+	ntpServer string
+	syncCmd   string
+	log       logging.Logger
+}
+
+func NewTimeSync(ntpServer string, syncCmd string) (*TimeSync, error) {
+	ts := &TimeSync{
+		ntpServer: ntpServer,
+		syncCmd:   syncCmd,
+		log:       logging.New("module", "time-sync"),
+	}
+
+	if _, err := ts.query(); err != nil {
+		return nil, err
+	}
+
+	return ts, nil
+}
+
+func (ts *TimeSync) Start() error {
+	ts.log.Debug("starting to sync time periodically")
+
+	if err := ts.sync(); err != nil {
+		return err
+	}
+
+	go func() {
+		for _ = range time.NewTicker(timeSyncInterval).C {
+			if synced, err := ts.checkNTPOffset(); err != nil {
+				ts.log.Error("failed to check ntp offset", "error", err)
+			} else if synced {
+				ts.log.Debug("time synced")
+				return
```

Comment:
Isn't it should be `continue` instead of `return`?

Reply:
I expect that the `TimeSync` runs continuously until the node shutdown.

Reply:
@kfangw Yes you right.